Automattic/wp-calypso
# Site Migration: add ability to store and restore progress

This PR stores the URL to local storage whenever the user progresses through the site migration flow.
In our email campaigns later we can leverage this to restore the progress of the user in the flow.

Related to #
pdDR7T-1sI-p2

## Proposed Changes

* Store the URL when site migration step changes
* Retrieve the stored URL and redirect when the `restore-progress` URL parameter is present 

## Testing Instructions
- Build the PR or use the image link
- In incognito window follow this flow /setup/import-hosted-site?ref=hosting-lp&section=performance
- Create a new site on tastewp
- Input that URL in the first step
- Go to the next step, check in the local storage if the URL is stored
- Open a new tab and visit: /setup/import-hosted-site?restore-progress
- The progress should be restored and you should be redirected to your last step





## Pre-merge Checklist
